tests/js/mock-use-instance-id.js

```diff
@@ -17,50 +17,74 @@
  */
 
 /**
- * External dependencies
+ * WordPress dependencies
  */
-import faker from 'faker';
+import { useMemoOne } from 'use-memo-one';
+import { useInstanceId } from '@wordpress/compose';
 
 /**
- * WordPress dependencies
+ * Creates a new ID for a given object.
+ *
+ * This uses a `Map` to track object instances, as opposed to the `WeakMap` used in the original `useInstanceId()` from @wordpress/compose.
+ * Using a regular `Map` to track objects will ensure they are not unpredictably garbage collected during tests, which can cause the
+ * generated instance IDs to change, with resulting test failures.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Map}    instanceMap Map of object instances to their current id.
+ * @param {Object} object      Object reference to create an id for.
+ * @return {number} The new id.
  */
-import { useMemo } from '@wordpress/element';
-import { useInstanceId } from '@wordpress/compose';
+function createID( instanceMap, object ) {
+	const instances = instanceMap.get( object ) || 0;
+	instanceMap.set( object, instances + 1 );
+
+	return instances;
+}
 
 /**
- * Provides a unique, memoized ID.
+ * Provides a unique instance ID.
  *
- * @since 1.107.0
+ * @since n.e.x.t
  *
- * @param {Object} object Object reference to create an ID for. This is unused and only included for compatibility with the original `useInstanceId()`.
- * @param {string} prefix Prefix for the unique ID.
- * @return {string} The unique ID.
+ * @param {Map}    instanceMap Map of object instances to their current id.
+ * @param {Object} object      Object reference to create an id for.
+ * @param {string} prefix      Prefix for the unique id.
+ * @return {string} The unique id.
  */
-function useMemoizedID( object, prefix ) {
-	return useMemo( () => {
-		const id = faker.datatype.uuid();
+function useInstanceID( instanceMap, object, prefix = '' ) {
+	return useMemoOne( () => {
+		const id = createID( instanceMap, object );
+
 		return prefix ? `${ prefix }-${ id }` : id;
-	}, [ prefix ] );
+	}, [ instanceMap, object, prefix ] );
 }
 
 /**
  * Mocks the `useInstanceId()` hook.
  *
  * This is necessary to ensure that the instance IDs generated by the hook are predictable during tests.
- * The original version of `useInstanceId()` in `@wordpress/compose` uses a `WeakMap` to track object references.
- * It appears that these can be garbage collected at different times during test runs on CI vs local, possibly
- * due to differing memory constraints. This results in different instance IDs being generated, which causes
- * instability in test snapshots.
  *
- * @since 1.107.0
+ * @since n.e.x.t
```

Comment:
When updating a function's `since` attribute, you should leave the original version, then add `n.e.x.t` with a comment about the updates.

See: https://github.com/google/site-kit-wp/blob/101917d33477c716d416a756628e7ad0871963d6/assets/js/googlesitekit/api/cache.js#L39-L41 as an example.

Reply:
Thanks for pointing that out, it seems I missed the override. Updated